### PR TITLE
Fixes for building OMPL Python bindings and FLANN extensions in modern environments

### PR DESCRIPTION
**Pull request by zkingston**
- Update C++ standard listed using `castxml`, added `-fsized-deallocation` to handle issue with gcc12
- Minor changes to FLANN cmake

### CODE REVIEW

File: CMakeModules/Findflann.cmake [hunk 1]
```diff
@@ -8,5 +8,20 @@ if(PKGCONFIG_FOUND)
     if(FLANN_LIBRARIES AND NOT FLANN_INCLUDE_DIRS)
         set(FLANN_INCLUDE_DIRS "/usr/include")
     endif()
+
+    if (FLANN_LIBRARY_DIRS)
+      foreach(FDIR ${FLANN_LIBRARY_DIRS})
+        string(REGEX MATCH "^-l" FOUND ${FDIR})
+        if (FOUND)
+          string(REPLACE "-l" "" FDIR2 ${FDIR})
```
Comment (mamoll) [inline CMakeModules/Findflann.cmake:16]:
```suggestion
          string(REGEX REPLACE "^-l" "" FDIR2 ${FDIR})
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -142,7 +142,7 @@ set_package_properties(flann PROPERTIES
 find_package(flann 1.8.3 QUIET)
 if (FLANN_FOUND)
     set(OMPL_HAVE_FLANN 1)
-    include_directories("${FLANN_INCLUDE_DIRS}")
+    include_directories(${FLANN_INCLUDE_DIRS})
```
Comment (mamoll) [inline CMakeLists.txt:145]: Once merged, you should make the corresponding change in `omplapp/CMakeLists.txt`.

**Review by mamoll — APPROVED** [review]

**Comment by mamoll** [comment]
I tested this on macOS 13.1. Dependencies installed via MacPorts. Works for me.

File: CMakeLists.txt (hunk 2 above)
Comment (zkingston) [reply CMakeLists.txt:145]: Will do!

**Comment by jacobjj** [comment]
I believe this also causes the py-bindings to fail on the [1.6.0 release](https://github.com/ompl/ompl/releases/tag/1.6.0)

**Comment by zkingston** [comment]
We will be releasing 1.6.1 soon, which includes this fix among others.

**Pull request merged**